modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AnalyticsTracker.kt

```diff
@@ -1,46 +1,22 @@
 package au.com.shiftyjelly.pocketcasts.analytics
 
-import au.com.shiftyjelly.pocketcasts.preferences.Settings
-
-object AnalyticsTracker {
-    private val trackers: MutableList<Tracker> = mutableListOf()
-    private lateinit var settings: Settings
-
-    fun init(settings: Settings) {
-        this.settings = settings
-        trackers.forEach { it.clearAllData() }
-    }
-
-    fun register(vararg trackers: Tracker) {
-        this.trackers.addAll(trackers)
-    }
-
+open class AnalyticsTracker constructor(
+    val trackers: List<Tracker>,
+    val isTrackingEnabled: () -> Boolean,
+) {
     fun track(event: AnalyticsEvent, properties: Map<String, Any> = emptyMap()) {
-        if (getSendUsageStats()) {
+        if (isTrackingEnabled()) {
             trackers.forEach { it.track(event, properties) }
         }
     }
 
-    fun refreshMetadata() {
-        trackers.forEach { it.refreshMetadata() }
-    }
+    fun refreshMetadata() = trackers.forEach(Tracker::refreshMetadata)
 
-    fun flush() {
-        trackers.forEach { it.flush() }
-    }
+    fun flush() = trackers.forEach(Tracker::flush)
 
-    fun clearAllData() {
-        trackers.forEach { it.clearAllData() }
-    }
+    fun clearAllData() = trackers.forEach(Tracker::clearAllData)
```

Comment:
As these methods don't return anything do they need to assign it back to the method? It makes me think it returns something when it doesn't. This is just personal preference so feel free to leave it.

Reply:
It's just a personal taste. I prefer inlining but I'll change open braces. It would be good to have a SCA rule to enfore it one way or the other.

Reply:
Fixed in 436bc4c

Reply:
A rule would be good or I don't mind not mentioning it again if it's just me.